src/Serialization/GAP.jl

```diff
@@ -0,0 +1,249 @@
+        prefix = GAP.getbangposition(Xnames, 1)::GapObj
+        save_object(s, string(prefix), :nameprefix)
+        names = Vector{String}(GAP.getbangposition(Xnames, 2)::GapObj)
+      else
+        # store the names
+        names = Vector{String}(Xnames)
+      end
+      save_data_array(s, :names) do
+        for i in names
+          save_object(s, i)
+        end
+      end
+      if !(GAP.Globals.HasIsWholeFamily(X) &&
+           GAPWrap.IsWholeFamily(X))
+        # store generators
+        save_data_array(s, :gens) do
+          for x in GAPWrap.GeneratorsOfGroup(X)
+            v = Vector{Int}(GAPWrap.ExtRepOfObj(x))
+            save_data_array(s) do
+              for i in v
+                save_object(s, i)
+              end
+            end
+          end
+        end
+      end
+    end
+  end)
+
+install_GAP_deserialization(:IsFreeGroup,
+  function(filt::GapObj, s::DeserializerState, T, d::Dict)
+    # We have to create a new free group.
+    wfilt = GAP.evalstr(d[:wfilt])
+    if haskey(d, :nameprefix)
+      # infinite rank
+#TODO: support these groups in Oscar?
+      prefix = string(d[:nameprefix])
+      init = GapObj(d[:names], true)
+      G = GAP.Globals.FreeGroup(wfilt, GAP.Globals.infinity, prefix, init)
+    else
+      names = GapObj(d[:names], true)
+      G = GAP.Globals.FreeGroup(wfilt, names)
+    end
+
+    if haskey(d, :gens)
+      generators = load_object(s, Vector, d[:gens], (Vector{Int}, Int))
+      F = GAPWrap.ElementsFamily(GAPWrap.FamilyObj(G))
+      Ggens = [GAPWrap.ObjByExtRep(F, GapObj(x, true)) for x in generators]
+      G = GAP.Globals.SubgroupNC(G, GapObj(Ggens))
+    end
+
+    return G
+  end)
+
+# # - IsElementOfFreeGroup
+# install_GAP_serialization(:IsElementOfFreeGroup, 
+#   function(X::GapObj, s::SerializerState)
+#     cfam = GAPWrap.FamilyObj(X)::GapObj
+#     efam = GAPWrap.ElementsFamily(cfam)::GapObj
+#     G = GAP.getbangproperty(efam, :freeGroup)
+#     save_data_dict(s) do
+#       save_object(s, G, :freeGroup)
+#     end
+#   end)
+# 
+# install_GAP_deserialization(:IsElementOfFreeGroup,
+#   function(filt::GapObj, s::DeserializerState, T, d::Dict)
+# 
+#   end)
+
+
+# - `IsSubgroupFpGroup`:
+#   full f.p. group or subgroup of it
+install_GAP_serialization(:IsSubgroupFpGroup, 
+  function(X::GapObj, s::SerializerState)
+    # save the defining data
+    elfam = GAPWrap.ElementsFamily(GAPWrap.FamilyObj(X))
+    save_data_dict(s) do
+      save_object(s, "IsSubgroupFpGroup", :GapType)
+
+      # underlying free group
+      freegroup = GAP.getbangproperty(elfam, :freeGroup)::GapObj
+      save_object(s, freegroup, :freeGroup)
+
+      # relators
+      relators = GAP.getbangproperty(elfam, :relators)::GapObj
+      save_data_array(s, :relators) do
+        for x in relators
+          v = Vector{Int}(GAPWrap.ExtRepOfObj(x))
+          save_data_array(s) do
+            for i in v
+              save_object(s, i)
+            end
+          end
```

Comment:
As above, can't we do this (and also in more places):
```suggestion
          save_object(s, v)
```

Reply:
Yes.
(I had copied this code from somewhere. Apparently the simplified code yields the same result.)